pandas/_typing.py

```diff
@@ -84,14 +84,14 @@
     NumpySorter = Optional[npt._ArrayLikeInt_co]  # type: ignore[name-defined]
 
     if sys.version_info >= (3, 10):
-        from typing import TypeGuard
+        from typing import TypeGuard  # pyright: ignore[reportUnusedImport]
```

Comment:
`as TypeGuard` would imho be nicer but not compatible with ruff https://github.com/astral-sh/ruff/issues/5698

Reply:
Pylint has the same rule (useless-import-alias)... Could go back to the pyright ignore if we really care about these two rules.